pkg/image/registryclient/client_mirrored.go

```diff
@@ -123,15 +123,17 @@ func (r *blobMirroredRepository) initialRepos(ctx context.Context) ([]reference.
 		return []reference.DockerImageReference{r.locator.ref}, false, nil
 	}
 
-	r.lock.Lock()
```

Comment:
This is concerning - why did you make this change?  This prevents multiple FirstRequests being called, so I didn't expect it to be removed.

Reply:
yeah, I think that's coming from the fact that Sally added `errorRepos` below but analyzing that again, it doesn't make sense, since we have higher-level methods responsible for calling `errorRepos`, I'll also update the comments with that information

Reply:
right, calling errorRepos from initialRepos there was a deadlock-there was limited context with use of FirstRequest, it's simply returned/not used w/ oc & image-registry, so i left the lock in errorRepos since that's where oc required it- wasn't foreseeing multiple FirstRequests - not calling errorRepos & returning the method's ref instead has the same result and is more appropriate.